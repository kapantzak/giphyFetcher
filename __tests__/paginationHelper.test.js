import {
  getPaginationItemRest,
  getPaginationItem,
  getNumberOfPages,
  getPaginationItemArrow,
  getPaginationArrow,
} from "../src/helpers/paginationHelper";

describe("getPaginationItemRest()", () => {
  test("Renders correctly", () => {
    const actual = getPaginationItemRest();
    expect(actual).toMatchSnapshot();
  });
});

describe("getPaginationItem()", () => {
  test("Renders correctly", () => {
    const callback = () => {};
    const actual = getPaginationItem(1, callback);
    expect(actual).toMatchSnapshot();
  });

  test("Active page rendered correctly", () => {
    const callback = () => {};
    const actual = getPaginationItem(1, callback, true);
    expect(actual).toMatchSnapshot();
  });
});

describe("getNumberOfPages()", () => {
  test.each([
    [1, 25, 25],
    [2, 25, 50],
    [2, 25, 26],
    [1, 25, 1],
  ])(
    "Returns %d pages for count %d of total %d",
    (expected, count, totalCount) => {
      const actual = getNumberOfPages({ count, totalCount });
      expect(actual).toBe(expected);
    }
  );
});

describe("getPaginationItemArrow()", () => {
  const callback = () => {};

  test("Left item renders correctly", () => {
    const actual = getPaginationItemArrow("left", callback);
    expect(actual).toMatchSnapshot();
  });

  test("Right item renders correctly", () => {
    const actual = getPaginationItemArrow("right", callback);
    expect(actual).toMatchSnapshot();
  });

  test("Returns null for invalid direction parameter", () => {
    const actual = getPaginationItemArrow("up", callback);
    expect(actual).toBeNull();
  });
});

describe("getPaginationArrow()", () => {
  test("Prev arrow renders correctly", () => {
    const actual = getPaginationArrow("left");
    expect(actual).toMatchSnapshot();
  });

  test("Next arrow renders correctly", () => {
    const actual = getPaginationArrow("right");
    expect(actual).toMatchSnapshot();
  });
});
