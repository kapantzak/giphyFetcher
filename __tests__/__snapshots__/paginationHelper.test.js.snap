// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPaginationArrow() Next arrow renders correctly 1`] = `
<i
  class="fas fa-angle-right"
/>
`;

exports[`getPaginationArrow() Prev arrow renders correctly 1`] = `
<i
  class="fas fa-angle-left"
/>
`;

exports[`getPaginationItem() Active page rendered correctly 1`] = `
<li
  class="page-item"
>
  <a
    class="page-link active"
    href="#"
  >
    1
  </a>
</li>
`;

exports[`getPaginationItem() Renders correctly 1`] = `
<li
  class="page-item"
>
  <a
    class="page-link"
    href="#"
  >
    1
  </a>
</li>
`;

exports[`getPaginationItemArrow() Left item renders correctly 1`] = `
<li
  class="page-item"
>
  <a
    class="page-link"
    href="#"
  >
    <i
      class="fas fa-angle-left"
    />
  </a>
</li>
`;

exports[`getPaginationItemArrow() Right item renders correctly 1`] = `
<li
  class="page-item"
>
  <a
    class="page-link"
    href="#"
  >
    <i
      class="fas fa-angle-right"
    />
  </a>
</li>
`;

exports[`getPaginationItemRest() Renders correctly 1`] = `
<li
  class="page-item"
>
  <span
    class="page-link"
  >
    ...
  </span>
</li>
`;
